### ISSUE TITLE
Hotplug on LSILogicSas sometimes fails

### ISSUE THREAD

**Issue by chrischdi** [coreos/bugs#2256]
# Issue Report #

### Container Linux Version ###

```
$ cat /etc/os-release
NAME="Container Linux by CoreOS"
ID=coreos
VERSION=1520.8.0
VERSION_ID=1520.8.0
BUILD_ID=2017-10-26-0342
PRETTY_NAME="Container Linux by CoreOS 1520.8.0 (Ladybug)"
ANSI_COLOR="38;5;75"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://issues.coreos.com"
COREOS_BOARD="amd64-usr"
```

### Environment ###

CoreOS running in VMWare ESXi, deployed using VMWare integrated Openstack.

### Expected Behavior ###

Volume is getting attached and is reachable in CoreOS somewhere in `/dev/...`.

### Actual Behavior ###

Volume is getting attached and is not reachable in CoreOS at `/dev/...`.
In `udevadm monitor` or `dmesg` you only see that it recognizes a SAS Port.
On reboot or possibly after detach and attach the disk is visible like expected.

### Reproduction Steps ###

  1. Boot CoreOS
  2. Attach LSILogicSAS controller
  3. Create Volume/Disk
  4. Attach Disk to LSILogicSAS controller
  5. If Disk is visible: detach disk and try to attach again

### Other Information ###

It works on a CentOS 7.3 which is using Kernel `3.10.0-514.26.2.el7.x86_64`.

It works on CoreOS when using the `paraVirtual` controller.

I tried to get more information about the events passed by the kernel to the mptsas module, which I think is the problem here.
So I did write `set linux_append="mptbase.mpt_debug_level=8"` to `/usr/share/oem/grub.cfg` to get more out of `dmesg`.
See the two attached files which show the messages on `dmesg` regarding `mpt` when attachment is working/not working.
[sas-not-working-dmesg.txt](https://github.com/coreos/bugs/files/1481285/sas-not-working-dmesg.txt)
[sas-working-dmesg.txt](https://github.com/coreos/bugs/files/1481286/sas-working-dmesg.txt)

I'm happy to provide more information if needed.

**Comment by chrischdi** [comment]
Further investigation/tests show the bug appears in Version 835.8.0 (Kernel 4.2.2).
Version 766.5.0 (Kernel 4.1.7) is working.

This problem may relate to the following:
[patchwork.kernel.org](https://patchwork.kernel.org/patch/9919973/)
[vmware kb](https://kb.vmware.com/s/article/2152054)
[suse announcment](https://www.suse.com/de-de/support/update/announcement/2017/suse-su-20172869-1/) (see mptsas)
[ubuntu launchpad](https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1730852)

**Comment by bgilbert** [comment]
This should be fixed in beta 1576.3.0 and stable 1520.9.0, due shortly.  Thanks for reporting.
